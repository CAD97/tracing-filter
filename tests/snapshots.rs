#![cfg(not(target_family = "wasm"))]
#![cfg(feature = "fancy-errors")]

use {
    std::{fs, path::Path},
    tracing_filter::{legacy, simple},
};

fn decolor(s: String) -> String {
    String::from_utf8(strip_ansi_escapes::strip(s).unwrap()).unwrap()
}

#[test]
fn snapshot_simple_filter_parser() {
    fn callback(path: &Path) {
        let src = fs::read_to_string(path).unwrap();
        match simple::Filter::parse(&src) {
            (Some(filter), Some(report)) => {
                let report = format!("{report:?}");
                insta::assert_snapshot!(
                    Some("simple"),
                    format!("{filter}\n\n{}", decolor(report)),
                    &src
                )
            },
            (Some(filter), None) => {
                insta::assert_snapshot!(Some("simple"), format!("{filter}\n(no warnings)"), &src)
            },
            (None, Some(report)) => {
                let report = format!("{report:?}");
                insta::assert_snapshot!(
                    Some("simple"),
                    format!("(compilation failed)\n\n{}", decolor(report)),
                    &src
                )
            },
            (None, None) => {
                insta::assert_snapshot!(Some("simple"), "(compilation failed)\n(no warnings)", &src)
            },
        }
    }

    insta::glob!("test_cases/*.env_filter", callback);
}

#[test]
fn snapshot_legacy_filter_parser() {
    fn callback(path: &Path) {
        let src = fs::read_to_string(path).unwrap();
        match legacy::Filter::parse(&src) {
            (filter, Some(report)) => {
                let report = format!("{report:?}");
                insta::assert_snapshot!(
                    Some("legacy"),
                    format!("{filter}\n\n{}", decolor(report)),
                    &src
                )
            },
            (filter, None) => {
                insta::assert_snapshot!(Some("legacy"), format!("{filter}\n\n(no warnings)"), &src)
            },
        }
    }

    insta::glob!("test_cases/*.env_filter", callback);
}
