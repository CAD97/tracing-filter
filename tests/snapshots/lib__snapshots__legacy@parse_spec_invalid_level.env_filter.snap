---
source: tests/snapshots.rs
expression: "crate1::mod1=noNumber,crate2=debug"
---
crate2=debug

1 directives were ignored as invalid

  ⚠ invalid level filter specified
   ╭────
 1 │ crate1::mod1=noNumber,crate2=debug
   ·              ────────
   ╰────
  help: valid level filters are OFF, ERROR, WARN, INFO, DEBUG, or TRACE


